pkg/envoy/util.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package envoy
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"istio.io/istio/pkg/test/env"
+)
+
+var (
+	envoyFileNamePattern = regexp.MustCompile("^envoy$|^envoy-[a-f0-9]+$|^envoy-debug-[a-f0-9]+$")
+)
+
+func findBinaries() ([]string, error) {
+	binPaths := make([]string, 0)
+	err := filepath.Walk(env.IstioOut, func(path string, f os.FileInfo, err error) error {
```

Comment:
this is a little suspicious.. do we need to do a walk? This basically looks like "just run any envoy binary you can find", which is rarely what we want. Can we restrict it to just search a folder non recursively maybe?

Reply:
The main motivation here is that we could have debug or release binaries, which are stored in different directories.

I suppose we could just search for release binaries?